x-pack/plugins/cases/server/routes/api/register_routes.ts

```diff
@@ -71,15 +71,18 @@ const logAndIncreaseDeprecationTelemetryCounters = ({
 };
 
 export const registerRoutes = (deps: RegisterRoutesDeps) => {
-  const { router, routes, logger, kibanaVersion, telemetryUsageCounter } = deps;
+  const { router, routes, logger, kibanaVersion, telemetryUsageCounter, isServerless } = deps;
 
   routes.forEach((route) => {
     const { method, path, params, options, routerOptions, handler } = route;
 
+    const routeAccess =
+      isServerless && options?.deprecated ? 'internal' : routerOptions?.access ?? 'internal';
```

Comment:
I see that `deprecated: true` can appear either in `options` or `routerOptions`. How about doing

```suggestion
      isServerless && ( options?.deprecated || routerOptions?.deprecated ) ? 'internal' : routerOptions?.access ?? 'internal';
```

Reply:
I thought about that and decided not to do it. The reason is that I thought that we should remove our `options.deprecated` in favor of the `routerOptions?.deprecated` which did not exist when we introduced our `options.deprecated`. Now that you mentioned it again I am not sure what is best and probably we should added for completeness. Wdyt?

Reply:
I get what we are doing here, and it saves a lot of work but maybe the decision of `isServerlessEnabled ? 'internal' : 'public'` should come down to the individual routes?

These are routes that are deprecated for some time but if suddenly we deprecate something else do we wanna make it immediately unavailable? Makes sense to have some migration period and this approach wouldn't let us do that in serverless.

Reply:
You are right. I will do it only for the selected routes.

Reply:
Sorry, I hadn't seen your previous message when I wrote mine the page was cached.

> I thought about that and decided not to do it. The reason is that I thought that we should remove our options.deprecated in favor of the routerOptions?.deprecated which did not exist when we introduced our options.deprecated. Now that you mentioned it again I am not sure what is best and probably we should do what you suggested for completeness. Wdyt?

i didn't know the reasoning for having both but if one is deprecated and the other is the preferred way and there is this uncertainty... I think this is another reason to be safer and just check for both with `||`

Reply:
Still, I prefer to do it only for the selected routes. You are right!

Reply:
Done in [`f4f33cc` (#198378)](https://github.com/elastic/kibana/pull/198378/commits/f4f33cc68c845b6434bd9bc8567d020c5f361539) and [`5648211` (#198378)](https://github.com/elastic/kibana/pull/198378/commits/5648211507c135545bb69e0a66f788053507872d)